src/hooks/useAmbassadors.tsx

```diff
@@ -156,6 +159,27 @@ export const AmbassadorsProvider = ({
       .then(setAmbassadors);
   }, []);
 
+  // Special refreshAmbassadors function with a delay
+  // Only used when !refresh is called by a privileged user
+  const refreshAmbassadors = useCallback(async () => {
+    return new Promise<void>((resolve) => {
+      setTimeout(
+        () => {
+          fetchAmbassadors()
+            .then(setAmbassadors)
+            .then(() => resolve())
+            .catch((err) => {
+              console.error(err);
+              resolve();
+            });
+        },
+        Math.floor(
+          Math.random() * (Math.floor(120) - Math.ceil(1) + 1) + Math.ceil(1),
+        ) * 1000,
+      );
+    });
+  }, []);
```

Comment:
Can we make the delay something that the logic that calls this method handles, rather than the method itself forcing the delay?